Objects/codeobject.c

```diff
@@ -549,6 +558,7 @@ PyCode_NewEmpty(const char *filename, const char *funcname, int firstlineno)
     struct _PyCodeConstructor con = {
         .filename = filename_ob,
         .name = funcname_ob,
+        .qualname = funcname_ob,
```

Comment:
Should we have `PyCode_NewEmpty` accept a new argument to set `qualname`?

Reply:
All the code object constructors is tricky business because this function was part of the stable API and the signature has changed considerably in 3.11, which was not technically allowed by our backwards compatibility policy, and we are pending a resolution.

My advice here is to not implement new APIs and try to default to old ones if the new parameters are not provided